### PR TITLE
Restructure documentation, port tutorials

### PR DESCRIPTION
**Pull request by dacoex**
This is the code for #120. Based on the pvsystem branch by @wholmgren 

The last commit now has a potential ToC as a list.

I think discussing using the list is easier and less effort.
Let's agree ToC level by ToC level.

To revewi, please also look at the RTD version:
[Contents (potential new ToC)](http://pvlib-python-dacoex.readthedocs.org/en/pvsystem_tutport/index.html#contents-potential-new-toc)

### CODE REVIEW

File: docs/sphinx/source/index.rst [hunk 1]
```diff
@@ -60,11 +60,54 @@ Contents
...
+
+* What’s New
+* Overview: Why pvlib?
+    * Introduction
+    * Modeling paradigms
+    * User extensions
+    * What it is not
+    * Comparison with PVLIB_MATLAB
+* Installation
+* Examples
```
Comment (wholmgren) [inline docs/sphinx/source/index.rst]: I'd remove the top level "Examples".

File: docs/sphinx/source/index.rst [hunk 2]
```diff
@@ -60,11 +60,54 @@ Contents
...
+        * irradiance.py tutorial 
+    * Specific datasets
+        * TMY tutorial 
+        * TMY data and diffuse irradiance model 
+        * TMY to Power Tutorial 
+    * System Modelling
+        * pvsystem tutorial
+        * Tracking
+        * Modeling multiple systems with system losses
+    * Validations
```
Comment (wholmgren) [inline docs/sphinx/source/index.rst]: I don't understand the "Validations" title. Are you planning to compare to real data? That sounds like a good idea, but the subheadings are not specific to data validation.

File: docs/sphinx/source/index.rst (hunk 1 above)
Comment (dacoex) [inline docs/sphinx/source/index.rst]: Was mimicked after
http://xarray.pydata.org/en/stable/index.html
http://xarray.pydata.org/en/stable/examples.html

File: docs/sphinx/source/index.rst (hunk 2 above)
Comment (dacoex) [inline docs/sphinx/source/index.rst]: Could be "Validations & Comparisons"

In the "SAPM" notebook, the different models are compared (= validated against each other).
This a strength of pvlib as compared to other tools albeit SAM & PVSyst can also do this...

File: docs/sphinx/source/index.rst (hunk 1 above)
Comment (wholmgren) [inline docs/sphinx/source/index.rst]: I see. I don't have a strong preference here.

File: docs/sphinx/source/index.rst (hunk 2 above)
Comment (wholmgren) [inline docs/sphinx/source/index.rst]: Ah, no, I would not use "validation" unless there was data.

Comment (dacoex) [inline docs/sphinx/source/index.rst]: So it will be "Validations & Comparisons"?

File: docs/sphinx/source/index.rst (hunk 1 above)
Comment (dacoex) [inline docs/sphinx/source/index.rst]: I suggest to keep the top level.
This helps to reduce `maxdepth` to 1 once there are many sub-pages.

File: docs/sphinx/source/index.rst (hunk 2 above)
Comment (wholmgren) [inline docs/sphinx/source/index.rst]: No, I suggest removing the section entirely and putting the subitems under system modeling. If you really want the section then call it something like "SAPM and Single Diode Comparison"

Comment (dacoex) [inline docs/sphinx/source/index.rst]: OK.

**Comment by dacoex** [comment]
So it seems that we are set with the ToC for now?

**Comment by wholmgren** [comment]
Yes, with the understanding that we'll probably want to change some of it once we start to see the content.

**Comment by dacoex** [comment]
Sure. Seems stepwise is a better approach....

**Comment by dacoex** [comment]
@wholmgren likly to pick this one up step by step.
Would you still agree to port the essential notebook based tutorials to python code / doctest so that they can be tested through an api restructuring as it happened with [v3.0](http://pvlib-python.readthedocs.io/en/latest/whatsnew.html#v0-3-0-march-21-2016)?

**Comment by wholmgren** [comment]
The documentation has grown since this conversation started. I would like to see more documentation pages similar to our Time and time zones, Clearsky, and Forecasting pages. Transposition and solar position pages would be relatively straightforward to write. I think a good ModelChain page is very important, but more work. I recommend closing this PR and doing these (or something like them that you care more about) one PR at a time.

**Comment by dacoex** [comment]
Thanks for responding.

Currently, my preference is with the pvsystem tutorial initiated by @jforbess

>I would like to see more documentation pages similar to our Time and time zones, Clearsky, and >Forecasting pages. 

This is what I was planning to convert the existing notebook into.

May question now:
**How do we ensure that the code in the documentation always up-to date?
Is it included in the tests?**

This shall be there to ensure that we do not damage the docs when refractoring the api of the core lib.

e.g. there are soem error in http://pvlib-python.readthedocs.io/en/latest/forecasts.html

_Some guidance would be appreciated before writing new pages._

Regarding ToC & structure, I see that in future there may be the followings technical pages:

```
- Fundamentals & Basics
   |---- Time and time zones
   |---- Clearsky
   |---- solar position
   |---- Transposition 
   |---- Input data
           |---- US TMY
           |---- Forecast & weather model (the whole section of Accessing Forecast Data to Weather Models deal with getting data)
           |---- other providers
           |---- local measurement & SCADA data (link to https://github.com/sandialabs/pecos )
- Modelling & Simulations
   |---- Trackers
   |---- Complete PV system
   |---- Sensitivity studies
   |---- Forecasting 
   |---- Cross-comparison & validation with other tools
   |---- etc.
- Analysis & Presentation
   |---- Statistisc ( ... should create a feature request issue because nothing there yet)
   |---- Plotting 
   |---- Reporting ( ... pvlib could be used to create a report like: 
                                http://solargis.com/assets/sample/SolarGIS-pvPlanner-sampleReport.xls
                                http://solargis.com/assets/sample/SolarGIS-pvPlanner-sampleReport.pdf) 
```

**Comment by wholmgren** [comment]
I've never written any doctests but I think you can do them in ipython directive. It will be a lot of work, so good luck. As I said before, go ahead and write up one new page in a new pr. 

I've explained the error on the forecasting page multiple times.

**Comment by dacoex** [comment]
>I've never written any doctests but I think you can do them in ipython directive. It will be a lot of >work, >so good luck. 

As alternative, we could use a separate code file being test & then use it as listing in the Spinx docs.

**Comment by wholmgren** [comment]
http://matplotlib.org/sampledoc/ipython_directive.html

```
.. ipython::

   In [1]: x = 'hello world'

   # this will raise an error if the ipython output is different
   @doctest
   In [2]: x.upper()
   Out[2]: 'HELLO WORLD'
```

**Comment by wholmgren** [comment]
There are some useful ideas in the discussion above, but the way to make progress at this point is starting and finishing one small PR before moving on to the next.

**Pull request closed without merging**